composer.json

```
@@ -104,6 +104,8 @@
         "phpunit/phpunit": "^8.5",
         "friendsofphp/php-cs-fixer": "^3.0",
         "squizlabs/php_codesniffer": "~3.0",
+        "phpcompatibility/php-compatibility": "*",
+        "phpcompatibility/phpcompatibility-joomla":"*",
```

```suggestion
        "phpcompatibility/phpcompatibility-joomla":"^2",
```

No need to `require-dev` PHPCompatibility, better to let PHPCompatibilityJoomla manage that as its that package' dependency.

Also, please be aware that PHPCompatibilityJoomla 2.x is based on PHPCompatibility 9.x, PHPCompatibilityJoomla 3.x will be based on PHPCompatibility 10.x (expected soonish). As that's a major, it will bring breaking changes (and some new dependencies), so you may want to do a controlled upgrade when the time comes.